vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml

```diff
@@ -259,7 +259,7 @@
       name: metal3clusters.infrastructure.cluster.x-k8s.io
       kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
     register: crd
-    failed_when: '"upgm3c" not in crd.resources[0].spec.names.shortNames'
+    failed_when: '"upgm3c" not in crd.resources[0].spec.names.shortNames and CAPM3RELEASE != CAPM3_REL_TO_VERSION'
```

Comment:
I think the version comparison should be  a `when` condition by itself  so that the assertion runs only when the versions are not the same